src/core/Directus/Filesystem/Files.php

```diff
@@ -406,13 +386,19 @@ public function saveEmbedData(array $fileInfo)
      *
      * @param string $path - file path
      * @param bool $outside - if the $path is outside of the adapter root path.
+     * @param string|null $localCopy - path to local copy of the file, if available
      *
      * @throws \RuntimeException
      *
      * @return array file information
      */
-    public function getFileInfo($path, $outside = false)
+    public function getFileInfo($path, $outside = false, $localCopy = null)
```

Comment:
shouldn't this become a reference? &$localCopy since we check for $tmp on the `saveData` function?

Also, shouldn't we check for $tmp with `is_null` instead of isset?

Reply:
On the reference, I don't think so. It should always be a string path to a file (if it's not null), and it isn't modified inside getFileInfo or getFileInfoFromLocalCopy.

Do you mean `!is_null`? We only want to run the `ffprobe` if `$tmp` is a string path to a file (I think). I added that early on because previously it was possible for `$tmp` [to be undefined](https://github.com/directus/api/blob/5a21686fdb7b456ec8c8999ac8b3ea372c6a55e3/src/core/Directus/Filesystem/Files.php#L321-L326) at that point in the function. The fact it never seems to have caused runtime errors makes me suspect that `saveData` is currently only being used for file uploads (i.e. with an `UploadedFile` for the `$fileData` argument) even though it looks like it was written with an eye toward supporting a second option where `$fileData` would be a string of some sort (see some of the comments referencing base64 data, which I didn't change).

I'm new to the codebase, though, so I'm not sure if that's accurate and I wanted to err on the side of making fewer changes if I could.

Reply:
I think we're good on this. Feel free to let me know of any other concerns.